### PR TITLE
fix(matrixify): apply dimension filters to all query-specific adhoc filter collections

### LINKED ISSUE #39007: 6.1.0rc1 - matrixify not applying to query B in Mixed Chart
### Bug description

Hello,

I am using Apache Superset 6.1.0rc1

I am using Mixed chart along with the matrixify feature activated with MATRIXIFY flag.
The dimension used to enable the vertical layout of matrixify only seems to apply to the data in Query A but not the data in query B.

### Screenshots/recordings

_No response_

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

_No response_

### Checklist

- [ ] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [ ] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

### PR DESCRIPTION
**Pull request by jenwitteng**
### SUMMARY

Matrixify builds a grid of chart cells, injecting a per-cell dimension filter (e.g. `country == 'USA'`) into each cell's `formData`. That filter was appended **only** to `adhoc_filters`.

Charts with more than one query route each query through a separate adhoc-filter collection — Mixed Chart reads query A from `adhoc_filters`, query B from `adhoc_filters_b`, query C from `adhoc_filters_c`. As a result, on multi-query charts the Matrixify slice filter was **silently dropped from every query except the first**, so those cells rendered unfiltered data.

This change introduces `appendMatrixifyFilters`, which fans the injected filters out to `adhoc_filters` **plus every `adhoc_filters_*` collection** present on the `formData`, so all of a chart's queries are sliced by the cell's dimension value.

Design notes:
- Non-array values on a matched key are treated as empty.
- Each key is reassigned to a **new** array rather than mutating the shared base `formData`, so cells don't contaminate each other.
- `matrixify_*` keys are deleted **before** the fan-out, so they can't be matched by the `adhoc_filters_*` pattern.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A — behavior fix, no visual/layout change. Effect: Mixed Chart Matrixify cells now filter query B/C data to the cell's dimension value instead of showing the full, unfiltered series.

### TESTING INSTRUCTIONS

Manual:
1. Create a Mixed Chart (query A + query B).
2. Enable Matrixify with a row dimension (e.g. `country` with 2+ values).
3. Before this change: query B's series ignore the per-cell dimension filter (every cell shows the same B data). After: both query A and query B are filtered to the cell's dimension value.

Automated:
- `superset-ui-core/.../Matrixify/MatrixifyGridGenerator.test.ts` — asserts the dimension filter reaches `adhoc_filters`, `adhoc_filters_b`, and `adhoc_filters_c` (including one that started empty), preserves pre-existing filters, and leaves look-alike keys such as `filters_b` untouched.
- `plugin-chart-echarts/test/MixedTimeseries/buildQuery.test.ts` — asserts both compiled queries carry the Matrixify filter and query B's pre-existing free-form filter is preserved.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #39007
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/chart/components/Matrixify/MatrixifyGridGenerator.test.ts [hunk 1]
```diff
@@ -116,6 +116,79 @@ test('should generate grid for dimensions mode', () => {
...
+  // separate collection. The cell's dimension filter must reach all of them.
+  const formDataWithMultipleQueries: TestFormData = {
+    viz_type: 'mixed_timeseries',
+    datasource: '1__table',
+    matrixify_enable: true,
+    matrixify_mode_rows: 'dimensions',
+    matrixify_mode_columns: 'disabled',
+    matrixify_dimension_rows: {
+      dimension: 'country',
+      values: ['USA'],
```
Comment (sadpandajoe) [inline superset-frontend/packages/superset-ui-core/src/chart/components/Matrixify/MatrixifyGridGenerator.test.ts]: With only one country, this test also passes if the helper appends in place to the shared `adhoc_filters_b` array; with USA and Canada cells, that would accumulate both country predicates in both cells and return no rows. Could this cover two countries and assert that each cell has only its own country filter and the original `adhoc_filters_b` remains unchanged?

Comment (jenwitteng) [reply superset-frontend/packages/superset-ui-core/src/chart/components/Matrixify/MatrixifyGridGenerator.test.ts]: yes added test to cover two dimension usa and canada.

**Review by rusackas — APPROVED** [review]
Thanks, @jenwitteng. LGTM! Sneaky bug, nice work tracking it down to the query-specific adhoc_filters_* collections. Also glad @sadpandajoe's two-country test case landed before merge, that was the real edge case here.
